modules/sor/sorV2/lib/pools/basePool.ts

```diff
@@ -10,5 +10,31 @@ export interface BasePool {
     getNormalizedLiquidity(tokenIn: Token, tokenOut: Token): bigint;
     swapGivenIn(tokenIn: Token, tokenOut: Token, swapAmount: TokenAmount, mutateBalances?: boolean): TokenAmount;
     swapGivenOut(tokenIn: Token, tokenOut: Token, swapAmount: TokenAmount, mutateBalances?: boolean): TokenAmount;
+    addLiquiditySingleTokenExactIn(
+        tokenIn: Token,
+        bpt: Token,
+        amount: TokenAmount,
+        mutateBalances?: boolean,
+    ): TokenAmount;
+    addLiquiditySingleTokenExactOut(
+        tokenIn: Token,
+        bpt: Token,
```

Comment:
Do you really need `bpt` as input here? Can you please review these interfaces to make sure they have only the inputs they really need?